wpilibj/src/main/java/edu/wpi/first/wpilibj/DutyCyclePotentiometer.java

```diff
@@ -0,0 +1,330 @@
+        this(channel, scale, 0);
+    }
+
+    /**
+     * AnalogPotentiometer constructor.
+     *
+     * <p>
+     * Use the fullRange and offset values so that the output produces meaningful
+     * values. I.E: you
+     * have a 270 degree potentiometer, and you want the output to be degrees with
+     * the starting point
+     * as 0 degrees. The scale value is 270.0(degrees).
+     *
+     * @param input The {@link DutyCycle} this potentiometer is plugged into.
+     * @param scale The scaling to multiply the voltage by to get a meaningful unit.
+     */
+    public DutyCyclePotentiometer(final DigitalSource input, double scale) {
+        this(input, scale, 0);
+    }
+
+    /**
+     * AnalogPotentiometer constructor.
+     *
+     * <p>
+     * Use the fullRange and offset values so that the output produces meaningful
+     * values. I.E: you
+     * have a 270 degree potentiometer, and you want the output to be degrees with
+     * the starting point
+     * as 0 degrees. The scale value is 270.0(degrees).
+     *
+     * @param input The {@link DutyCycle} this potentiometer is plugged into.
+     * @param scale The scaling to multiply the voltage by to get a meaningful unit.
+     */
+    public DutyCyclePotentiometer(final DutyCycle input, double scale) {
+        this(input, scale, 0);
+    }
+
+    /**
+     * AnalogPotentiometer constructor.
+     *
+     * <p>
+     * The potentiometer will return a value between 0 and 1.0.
+     *
+     * @param channel The analog input channel this potentiometer is plugged into.
+     *                0-3 are on-board
+     *                and 4-7 are on the MXP port.
+     */
+    public DutyCyclePotentiometer(final int channel) {
+        this(channel, 1, 0);
+    }
+
+    /**
+     * AnalogPotentiometer constructor.
+     *
+     * <p>
+     * The potentiometer will return a value between 0 and 1.0.
+     *
+     * @param input The {@link DutyCycle} this potentiometer is plugged into.
+     */
+    public DutyCyclePotentiometer(final DigitalSource input) {
+        this(input, 1, 0);
+    }
+
+    /**
+     * AnalogPotentiometer constructor.
+     *
+     * <p>
+     * The potentiometer will return a value between 0 and 1.0.
+     *
+     * @param input The {@link DutyCycle} this potentiometer is plugged into.
+     */
+    public DutyCyclePotentiometer(final DutyCycle input) {
+        this(input, 1, 0);
+    }
+
+    /**
+     * Get the current reading of the potentiometer.
+     *
+     * @return The current position of the potentiometer.
+     */
+    public double get() {
+        if (m_simPosition != null) {
+            return m_simPosition.get();
+        }
+        double pos;
+        if (m_periodNanos == 0.0) {
+            pos = m_dutyCycle.getOutput();
+        } else {
+            int highTime = m_dutyCycle.getHighTimeNanoseconds();
+            pos = highTime / m_periodNanos;
+        }
+
+        return pos * m_fullRange + m_offset;
```

Comment:
Could this be wrapped back into (0, m_fullRange) (e.g. via MathUtil.inputModulus)?
This is an issue with DutyCycleEncoder I finally figured out this year- we were expecting 0-1 outputs from getAbsolutePosition but instead got (-offset, 1-offset) which led to inconsistent behavior depending on where the mechanism started/where the encoder's zero position was.